### PR TITLE
Bringing Message, MessageType, tokens and config up to date

### PR DESCRIPTION
**Pull request by chx**
Quick easy cleanup of deprecated and removed stuff. This doesn't pass Travis yet but it is definitely a huge step forward. Only MessageUiTest fails and the way I avoid token double escaping is using a `@internal` method but what can I do? I filed https://www.drupal.org/node/2662296

### CODE REVIEW

File: src/Entity/MessageType.php [hunk 1]
```diff
@@ -323,10 +327,13 @@ public function getUuid() {
   public function getText($langcode = NULL, $options = array()) {
     $text = $this->text;
 
-    if ($langcode && \Drupal::moduleHandler()->moduleExists('config_translation')) {
-      $config_translation = \Drupal::languageManager()->getLanguageConfigOverride($langcode, 'message.type.' . $this->id());
-      if ($translated_text = $config_translation->get('text')) {
-        $text = $translated_text;
+    if ($langcode) {
+      $language_manager = \Drupal::languageManager();
+      if ($language_manager instanceof ConfigurableLanguageManagerInterface) {
```
Comment (amitaibu) [inline src/Entity/MessageType.php:332]: why is this needed?

Comment (chx) [inline src/Entity/MessageType.php]: Because at least testing is easier  and conceptually, you do not care really how the getLanguageConfigOverride gets there, it violates encapsulation to check for a module. You want *something* to provide getLanguageConfigOverride and don't care about anything else.

Comment (amitaibu) [inline src/Entity/MessageType.php:332]: oh this is instead of the `moduleExists` - sorry, I overlooked it. Yap, makes more sense :)

Comment (chx) [inline src/Entity/MessageType.php]: If it's not clear: by default \Drupal::languageManager() is a LanguageManagerInterface not a ConfigurableLanguageManagerInterface.

Comment (chx) [inline src/Entity/MessageType.php]: Yes this remove moduleExist, doesn't do anything else.

Comment (chx) [inline src/Entity/MessageType.php]: If there's nothing else I'll go to sleep now :)

Comment (amitaibu) [inline src/Entity/MessageType.php:332]: Don't let me stop you from a good sleep. Anyway, It's @RoySegall that pushes the merge for the 8.x version ;)

Comment (chx) [inline src/Entity/MessageType.php]: Let's hope then he merges this soon :) I will be back with UI and more token fixes.

Comment (RoySegall) [inline src/Entity/MessageType.php:332]: Looks a lot more cleaner. There are couple of more things I need to do before merging it but i'll merge it.

Comment (RoySegall) [inline src/Entity/MessageType.php:332]: And oh yeah - 10x for the work.
